examples/server/utils.hpp

```diff
@@ -14,6 +14,17 @@
 
 using json = nlohmann::json;
 
+// https://community.openai.com/t/openai-chat-list-of-error-codes-and-types/357791/11
+enum error_type {
+    ERROR_INVALID_REQUEST,
+    ERROR_AUTHENTICATION,
+    ERROR_SERVER,
+    ERROR_NOT_FOUND,
+    ERROR_PERMISSION,
+    ERROR_UNAVAILABLE, // custom error
+    ERROR_NOT_SUPPORTED, // custom error
+};
+
```

Comment:
Enum values always use the full prefix of the enum name:

```suggestion
enum error_type {
    ERROR_TYPE_INVALID_REQUEST,
    ERROR_TYPE_AUTHENTICATION,
    ERROR_TYPE_SERVER,
    ERROR_TYPE_NOT_FOUND,
    ERROR_TYPE_PERMISSION,
    ERROR_TYPE_UNAVAILABLE, // custom error
    ERROR_TYPE_NOT_SUPPORTED, // custom error
};

```